Mage.Sets/src/mage/cards/d/DesertWarfare.java

```diff
@@ -0,0 +1,225 @@
+                .addHint(DesertsYouControlCount.getHint()));
+    }
+
+    private DesertWarfare(final DesertWarfare card) {
+        super(card);
+    }
+
+    @Override
+    public DesertWarfare copy() {
+        return new DesertWarfare(this);
+    }
+
+}
+
+//Based on PutCardIntoGraveFromAnywhereAllTriggeredAbility
+class DesertWarfareTriggeredAbility extends TriggeredAbilityImpl {
+
+    private static FilterOwnedCard filterCard = new FilterOwnedCard();
+
+    static {
+        filterCard.add(SubType.DESERT.getPredicate());
+    }
+
+    private final SetTargetPointer setTargetPointer;
+
+    public DesertWarfareTriggeredAbility(Zone zone, Effect effect, boolean optional, TargetController targetController, SetTargetPointer setTargetPointer) {
+        super(zone, effect, optional);
+        FilterOwnedCard filter = filterCard.copy();
+        this.setTargetPointer = setTargetPointer;
+        filter.add(targetController.getOwnerPredicate());
+        setTriggerPhrase("Whenever you sacrifice a Desert and whenever a Desert card is put into your graveyard from your hand or library, ");
+    }
+
+    protected DesertWarfareTriggeredAbility(final DesertWarfareTriggeredAbility ability) {
+        super(ability);
+        filterCard = filterCard.copy();
+        this.setTargetPointer = ability.setTargetPointer;
+    }
+
+    @Override
+    public DesertWarfareTriggeredAbility copy() {
+        return new DesertWarfareTriggeredAbility(this);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.ZONE_CHANGE
+                || event.getType() == GameEvent.EventType.SACRIFICED_PERMANENT;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        Card card = game.getCard(event.getTargetId());
+        if (card == null) {
+            return false;
+        }
+        switch (event.getType()) {
+            case SACRIFICED_PERMANENT:
+                Permanent permanent = game.getPermanentOrLKIBattlefield(event.getTargetId());
+                if (permanent == null || !permanent.hasSubtype(SubType.DESERT, game)) {
+                    return false;
+                }
+                break;
+            case ZONE_CHANGE:
+                ZoneChangeEvent zEvent = (ZoneChangeEvent) event;
+                if (zEvent.getToZone() != Zone.GRAVEYARD
+                        || (zEvent.getFromZone() != Zone.LIBRARY && zEvent.getFromZone() != Zone.HAND)
+                        || !filterCard.match(card, getControllerId(), this, game)) {
+                    return false;
+                }
+                break;
+            default:
+                return false;
+        }
+        this.getEffects().setTargetPointer(new FixedTarget(event.getTargetId(), game));
+        return true;
+    }
+}
+
+class DesertWarfareReturnEffect extends OneShotEffect {
```

Comment:
why do you actually need a custom class, won't `ReturnFromGraveyardToBattlefieldTargetEffect` suffice?